homeassistant/components/light/__init__.py

```diff
@@ -241,7 +241,20 @@ def async_handle_light_service(service):
         if color_name is not None:
             params[ATTR_RGB_COLOR] = color_util.color_name_to_rgb(color_name)
 
+        xy_color = params.get(ATTR_XY_COLOR, None)
+
         for light in target_lights:
+            # Convert xy_color if unsupported
+            if xy_color is not None:
+                support_xy = bool(light.supported_features & SUPPORT_XY_COLOR)
+                brightness = params.get(ATTR_BRIGHTNESS, light.brightness)
+
+                if not support_xy and brightness is not None:
+                    rgb = color_util.color_xy_brightness_to_RGB(
```

Comment:
It does, if you really want to enforce a single user-provided ```brightness``` value. The current code uses a per-light ```brightness``` and thus cannot be moved out of the loop.